### ISSUE TITLE
@ArgumentCollection should accept a disambiguation prefix.

### ISSUE THREAD

**Issue by vruano** [broadinstitute/gatk#2582]
Currently I found myself in a scenario in where it would be great to have the ability of declaring a second set of interval-arguments for compose a separated interval list like so: 

```
java -jar gatk.jar MyTool -inputL intervals.list -targetL targetIntervals.list 
```

Ideally I would like to be able to reuse the interval parsing engine in ```IntervalArgumentCollection``` for both sets but that is not possible due to argument name collisions. 

What I would be able to do is this:

```
     @ArgumentCollection(prefix="input")
    public IntervalArgumentCollection inputIntervals = ...;
   ...
    @ArgumentCollection(prefix="target")
   public IntervalArgumentCollection inputIntervals = ...;
```

**Comment by lbergelson** [comment]
That seems like a very reasonable thing.  It would also provide a reasonable way to build complex command lines from a chain of spark tools.  It's like the picard "nested command line parsers" but gives explicit control to the tool author.  One wrinkle is that it would have to work recursively with nested argument collections, but I don't see any problem there.   

@cmnbroad What do you think?

**Comment by cmnbroad** [comment]
Yeah, sounds very similar to NestedOptions. I added a ticket to Barclay.
